### PR TITLE
feat: add sha256sums for kind binaries

### PR DESCRIPTION
**Pull request by viveksyngh**
- Add sha256sum file for kind binaries build as a part of release process

Fixes: #2331

### CODE REVIEW

File: hack/release/build/cross.sh [hunk 1]
```diff
@@ -48,3 +48,6 @@ export GOOS=linux GOARCH=arm64
 export GOOS=linux GOARCH=ppc64le
 EOF
 )
+
+# add sha256 for binaries
+for f in ${REPO_ROOT}/bin/kind-*; do shasum -a 256 $f > $f.sha256sum; done
```
Comment (xinydev) [inline hack/release/build/cross.sh:53]: the original output for `shasum` is something like `hashValue  fileName`, I think it only needs the hash value here?  

https://github.com/kubernetes/kompose/blob/0b331d9e5dede975d62ca382b2dd7dff89019f6e/script/release.sh#L243

Comment (aojea) [reply hack/release/build/cross.sh:53]: how do you know the filaname to check it with the command?

```
[aojea@aojea-laptop ~]$ shasum -a 256 ifac.go > ifac.sha
[aojea@aojea-laptop ~]$ sha256sum -c ifac.sha 
ifac.go: OK
[aojea@aojea-laptop ~]$ cp ifac.sha ifac.shaonly
[aojea@aojea-laptop ~]$ vi ifac.shaonly #removed the filename
[aojea@aojea-laptop ~]$ sha256sum -c ifac.shaonly 
sha256sum: ifac.shaonly: no properly formatted SHA256 checksum lines found
```

**Comment by aojea** [comment]
the verify job fails 
```
In ./hack/release/build/cross.sh line 53:
for f in ${REPO_ROOT}/bin/kind-*; do shasum -a 256 $f > $f.sha256sum; done
         ^----------^ SC2231: Quote expansions in this for loop glob to prevent wordsplitting, e.g. "$dir"/*.txt .
                                                   ^-- SC2086: Double quote to prevent globbing and word splitting.
                                                        ^-- SC2086: Double quote to prevent globbing and word splitting.
```

File: hack/release/build/cross.sh (hunk 1 above)
Comment (xinydev) [reply hack/release/build/cross.sh:53]: You are right, it is not easy to check.  

however,I found that we have a lot of current tar files are only have the hash value,I'm just curious what is the reason for this.

`
➜  ~ gsutil cat gs://kubernetes-release/release/v1.21.0-alpha.0/kubernetes-test-windows-amd64.tar.gz.sha256   
e3b8a57d0bba4665f54874e4478bf2b360254314a482f4ba36f15ce228c561b7%     

➜  ~ gsutil cat gs://kubernetes-release/release/v1.21.0-alpha.0/kubernetes-src.tar.gz.sha256   
7eaf8fa1aa03afead4292d1e10dbc839439b4cd45a5a749bac242f8cf0040f45%    
`

Comment (aojea) [reply hack/release/build/cross.sh]: not an expert here, I always used the hash filename format, that is what shasum reports directly

File: hack/release/build/cross.sh [hunk 2]
```diff
@@ -48,3 +48,6 @@ export GOOS=linux GOARCH=arm64
 export GOOS=linux GOARCH=ppc64le
 EOF
 )
+
+# add sha256 for binaries
+for f in "${REPO_ROOT}"/bin/kind-*; do shasum -a 256 "$f" > "$f".sha256sum; done
```
Comment (aojea) [inline hack/release/build/cross.sh:53]: don't use the oneliner and keep the last line, is easier to read
```suggestion
for f in "${REPO_ROOT}"/bin/kind-*; do 
    shasum -a 256 "$f" > "$f".sha256sum
done

```

Comment (viveksyngh) [reply hack/release/build/cross.sh]: Updated

Comment (aojea) [reply hack/release/build/cross.sh]: hmm this adds the whole path to the generated file, I think that we have to cd first in the folder
```
cd "${REPO_ROOT}"/bin
for f in kind-*; do
      shasum -a 256 "$f" > "$f".sha256sum
done
```

Comment (viveksyngh) [reply hack/release/build/cross.sh]: yup it was adding full path. I have updated that to change to the directory and then run sha command.

File: hack/release/build/cross.sh (hunk 1 above)
Comment (viveksyngh) [reply hack/release/build/cross.sh]: I have also used hash filename format for one of the CLI in the past. Not sure what benefit other approach has.

**Comment by aojea** [comment]
/lgtm
thanks
/assign @BenTheElder

File: hack/release/build/cross.sh [hunk 3]
```diff
@@ -48,3 +48,9 @@ export GOOS=linux GOARCH=arm64
 export GOOS=linux GOARCH=ppc64le
 EOF
 )
+
+# add sha256 for binaries
+cd "${REPO_ROOT}"/bin
```
Comment (BenTheElder) [inline hack/release/build/cross.sh:53]: TODO: refactor to not need `cd`
TODO: update loop to ignore files with .sha256sum extension.

**Review by BenTheElder — APPROVED** [review]
/lgtm
/approve
/hold
thanks!

**Comment by BenTheElder** [comment]
/hold cancel
we can iterate later, thank you!

**Pull request merged**